File: backend/app/schemas/job.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from uuid import UUID


class JobSourceOut(BaseModel):
    id: UUID
    source: str
    url: str

    model_config = {"from_attributes": True}


class JobOut(BaseModel):
    id: UUID
    title: str
    company: str
    location: Optional[str] = None
    experience: Optional[str] = None
    employment_type: Optional[str] = None
    is_active: bool
    crawled_at: datetime
    sources: List[JobSourceOut] = []

    model_config = {"from_attributes": True}


class JobListResponse(BaseModel):
    jobs: List[JobOut]
    total: int
    page: int
    size: int
    has_next: bool

File: frontend/src/types/job.ts
export interface JobSource {
  id: string;
  source: "wanted" | "jumpit" | "programmers" | "catch" | "groupby";
  url: string;
}

export interface Job {
  id: string;
  title: string;
  company: string;
  location: string | null;
  experience: string | null;
  employment_type: string | null;
  is_active: boolean;
  crawled_at: string;
  sources: JobSource[];
}

export interface JobListResponse {
  jobs: Job[];
  total: number | null;
  page: number;
  size: number;
  has_next: boolean;
}

export type CardStatus =
  | "unread"
  | "viewed"
  | "last_seen"
  | "saved"
  | "applied"
  | "rejected";

export interface SavedJob extends Job {
  status: "saved" | "applied" | "rejected";
  savedAt: number;
}

export interface FilterState {
  q: string;
  location: string[];
  experience: string[];
  employment_type: string[];
  source: string[];
}
